main/src/library/Console.flix

```diff
@@ -91,6 +91,41 @@ namespace Console {
                 esc + "[38;2;1m" + ToString.toString(s) + esc + "[0m"
         }
 
+    ///
+    /// Returns `s` formatted with color `hexCode`.
+    /// Expected format of `hexCode` example: "#CC33AE"
+    ///
+    pub def hex(hexCode: String, s: a): String with ToString[a] =
+        match String.stripPrefix(substr = "#", hexCode) {
+            case Some(hexNumbers) => {
+                if (String.length(hexNumbers) == 6) {
+                    let red = String.slice(0, 2, hexNumbers);
```

Comment:
This is better, but I think it can still be improved:
1. IMO, barring exceptional circumstances, Flix functions should _never_ be expected to throw exceptions. So `stringToHex` should catch the exception itself and return an `Option` to indicate success.
2. We return a string containing an error message below, which isn't part of the spec of the function. Since the spec is now "if there's an error, return the original string", we should do that in the case of _any_ possible error.
3. Since there are several possible points of error, this is a likely opportunity for the `let*` syntax. (see [here](https://doc.flix.dev/tipstricks/) if you're not familiar)

Let me know if any of that is unclear!

Reply:
NB: Matt is not angry or commanding, just passionate :)

To rephrase:

1. I think `stringToHex` as a helper could return `Option` and then inside `hex` we just unwrap it and if it is `None` return the same string `s`.
2. Yes, I should have said "return the input string s".

Reply:
Also I think we often fall into the trap of giving too little feedback because we are rushed. But we should remember that many things are not well described. Hopefully this improves with api.flix.dev and so forth :)

Reply:
I think it is nice to get some feedback! Then I learn to write better code :-)